GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -956,10 +959,34 @@ private async Task UpdateBranchNameDisplayAsync()
             await TaskScheduler.Default;
 
             var currentBranchName = Module.GetSelectedBranch();
+            if (_branchNameLabelOnClick != null)
+            {
+                remoteNameLabel.Click -= _branchNameLabelOnClick;
+            }
+
+            var currentBranch = Module.GetRefs(false, true).FirstOrDefault(r => r.LocalName == currentBranchName);
+            if (currentBranch == null)
+            {
+                branchNameLabel.Text = currentBranchName;
+                remoteNameLabel.Text = string.Empty;
+                return;
+            }
+
+            var pushTo = string.IsNullOrEmpty(currentBranch.TrackingRemote) || string.IsNullOrEmpty(currentBranch.MergeWith)
```

Comment:
I'm finding the condition is too long to grasp it easily
Do you think we we can present it in a more traditional way - `if..else`?